src/Database/LSMTree/Internal/Chunk.hs

```diff
@@ -19,23 +20,31 @@ import           Prelude hiding (length)
 
 import           Control.Exception (assert)
 import           Control.Monad.ST.Strict (ST)
+import           Data.ByteString (ByteString)
 import           Data.List (scanl')
 import           Data.Primitive.PrimVar (PrimVar, newPrimVar, readPrimVar,
                      writePrimVar)
-import           Data.Vector.Primitive (Vector, length, unsafeCopy,
+import           Data.Vector.Primitive (Vector (Vector), length, unsafeCopy,
                      unsafeFreeze)
 import           Data.Vector.Primitive.Mutable (MVector)
 import qualified Data.Vector.Primitive.Mutable as Mutable (drop, length, slice,
                      take, unsafeCopy, unsafeNew)
 import           Data.Word (Word8)
+import           Database.LSMTree.Internal.ByteString (byteArrayToByteString)
 
 -- * Chunks
 
 -- | A chunk of bytes, typically output during incremental index serialisation.
-newtype Chunk = Chunk (Vector Word8)
+newtype Chunk = Chunk (Vector Word8) deriving stock (Eq, Show)
 
-fromChunk :: Chunk -> Vector Word8
-fromChunk (Chunk content) = content
+-- | Yields the contents of a chunk as a byte vector.
+toByteVector :: Chunk -> Vector Word8
+toByteVector (Chunk byteVector) = byteVector
+
+-- | Yields the contents of a chunk as a byte string.
```

Comment:
```suggestion
-- | Yields the contents of a chunk as a strict byte string.
```

Reply:
I made a corresponding change in 9b1599ee1eaf72ce29a4bbe7f84d191b7f47fb2f. I put the “strict” in parentheses, since the naming within the `bytestring` package and also the naming of the above function is centered around the idea that strict byte strings are the default.